api/notd/collection_processor.py

```diff
@@ -59,28 +63,41 @@ async def retrieve_collection(self, address: str) -> RetrievedCollection:
             collectionSymbol = tokenMetadataSymbolResponse[0]
         except BadRequestException:
             collectionSymbol = None
-        openseaResponse = None
-        retryCount = 0
-        while not openseaResponse:
-            try:
-                openseaResponse = await self.requester.get(url=f'https://api.opensea.io/api/v1/asset_contract/{address}', headers={"X-API-KEY": self.openseaApiKey})
-            except ResponseException as exception:
-                if exception.statusCode == 404:
-                    raise CollectionDoesNotExist()
-                if retryCount >= 3 or (exception.statusCode < 500 and exception.statusCode != 429):
-                    break
-                logging.info(f'Retrying due to: {str(exception)}')
-                await asyncio.sleep(1.5)
-            except ReadTimeout as exception:
-                if retryCount >= 3:
-                    break
-                logging.info(f'Retrying due to: {str(exception)}')
-                await asyncio.sleep(1.5)
-            retryCount += 1
-        openseaCollection = openseaResponse.json().get('collection') if openseaResponse else {}
-        if openseaCollection is None:
-            logging.info(f'Failed to load collection from opensea: {address}')
-            openseaCollection = {}
+        try:
+            contractUriResponse = await self.ethClient.call_function(toAddress=address, contractAbi=self.erc721MetdataContractAbi, functionAbi=self.erc721MetadataContractUriFunctionAbi)
+            contractMetadataUri = contractUriResponse[0]
+        except BadRequestException:
+            contractMetadataUri = None
+        if contractMetadataUri is not None:
+            contractMetadataUriResponse = await self.requester.get(url=contractMetadataUri)
+            openseaCollection = contractMetadataUriResponse.json()
+            if isinstance(openseaCollection, str):
+                openseaCollection = json.loads(openseaCollection)
+            print(openseaCollection)
+            await self.s3manager.write_file(content=str.encode(json.dumps(openseaCollection)), targetPath=f'{self.bucketName}/collection-metadatas/{address}/{date_util.datetime_from_now()}.json')
+        else:
+            openseaResponse = None
```

Comment:
instead of doing this as either the metadataURI or opensea we should do both. So do it like this:
1. get metadataURI. if its there populate its fields into metadata in a purposeful way e.g.
```
collectionName = collectionName or collectionMetadata.get('name')
collectionDescription = collectionDescription or collectionMetadata.get('description')
collectionImageUrl = collectionMetadata.get('image') or collectionMetadata.get('<another possible image field>')
... other fields ...
```
2. get opensea and do the same as above but with openseaCollection instead